src/psyclone/transformations.py

```diff
@@ -1252,6 +1251,148 @@ def apply(self, target_nodes, options=None):
 
         return schedule, keep
 
+class OMPSingleTrans(ParallelRegionTrans):
+    '''
+    Create an OpenMP SINGLE region by inserting directives. The most
+    likely use case for this transformation is to wrap around task-based
+    transformations. The parent region for this should usually also be
+    a OMPParallelTrans. For example:
+
+    >>> from psyclone.parse.algorithm import parse
+    >>> from psyclone.parse.utils import ParseError
+    >>> from psyclone.psyGen import PSyFactory
+    >>> from psyclone.errors import GenerationError
+    >>> api = "gocean1.0"
+    >>> filename = "nemolite2d_alg.f90"
+    >>> ast, invokeInfo = parse(filename, api=api, invoke_name="invoke")
+    >>> psy = PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.transformations import OMPParallelTrans, OMPSingleTrans
+    >>> singletrans = OMPSingleTrans()
+    >>> paralleltrans = OMPParallelTrans()
+    >>>
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # SINGLE region
+    >>> singletrans.apply(schedule.children)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # PARALLEL region
+    >>> paralleltrans.apply(schedule.children)
+    >>> schedule.view()
+
+    '''
+    # The types of node that this transformation cannot enclose
+    excluded_node_types = (nodes.CodeBlock, nodes.Return, nodes.ACCDirective,
+                           psyGen.HaloExchange)
+
+    def __init__(self, nowait=False):
+        super(OMPSingleTrans, self).__init__()
+        # Set the type of directive that the base class will use
+        self._pdirective = self._directive
+        #Store whether this single directive has a barrier or not
+        self._omp_nowait = nowait
+
+    def __str__(self):
+        return "Insert an OpenMP Single region"
+
+    @property
+    def omp_nowait(self):
+        ''' Returns whether or not this Single region uses a nowait
+            clause to remove the end barrier. The default value is False
+        '''
+        return self._omp_nowait
+
+    @property
+    def name(self):
+        '''
+        :returns: the name of this transformation as a string.
+        :rtype: str
+        '''
+        return "OMPSingleTrans"
+
+    @omp_nowait.setter
+    def omp_nowait(self, value):
+        ''' Sets the nowait property that will be specified by
```

Comment:
Please add a :param: for the `value` argument and a :raises: for the TransformationError. In fact, I think the TransformationError ought to be a TypeError really.

Reply:
Done